### PR TITLE
Operating Console Buffs

### PR DESCRIPTION
**Pull request by VisVirific**
* Operating Consoles can now display medical scans inserted into them, being capable of updating it as desired.
* Operating Consoles will now sound off specific alarms and warnings for some specific conditions.
* They also show Blood Volume too.
* Medical scans now are their own unique paper type, `medscan`! With added unique sprites to help discern it from other papers.
* Operating Table now has an additional icon for when patients flatline

**Most changes in a nutshell**

**Operating Table Flatline**

### CODE REVIEW

**Comment by VisVirific** [comment]
!review

**Review by mikomyazaki — COMMENTED** [review]
* When I open the computer UI on my game, it is extremely small. Looks like it is limited to 200x200px in the code which is too small to see everything. 
* Is it intentional that the operating table cannot produce its own medscan report item, unless it has already been given one printed from the advanced medical scanner? You could make this clearer in the changelog or somewhere in-game, like "Insert Health Analyzer Report" instead of "Patient Scan not found." maybe?
* From a quick check, it looks like the advanced medical scanner only spawns in the CMOs closet, is that right? If so, does it make sense that only the CMO will be able to use this feature on the operating tables?
* On the operating table graphic, there is a little green animated pulse display. Could you make that change colour to red / whatever, when there is an audio alert condition? Currently it just stays the same even if the patient is dead.
* The alarm text gets kinda spammy, and the chat spam prevention doesn't help because they are two different messages. You could change it to "Warning! x's Brain Activity and Blood Oxygenation are in critical condition." when both are true? Maybe increase the delay between messages (not necessarily the beeps aswell) too?

**Comment by VisVirific** [comment]
>* When I open the computer UI on my game, it is extremely small. Looks like it is limited to 200x200px in the code which is too small to see everything.
Hell, oversight. I had changed the console UI to be larger to open at the appropriate size like in the gift. Looks like I forgot to change it in the github

>* Is it intentional that the operating table cannot produce its own medscan report item, unless it has already been given one printed from the advanced medical scanner? You could make this clearer in the changelog or somewhere in-game, like "Insert Health Analyzer Report" instead of "Patient Scan not found." maybe?
Yes, it's intentional so the Operating Console isn't just another form of Body Scanner, bypassing the need for a initial scan. I will make it clearer though, and also change the "Patient Scan not found" message accordingly

>* From a quick check, it looks like the advanced medical scanner only spawns in the CMOs closet, is that right? If so, does it make sense that only the CMO will be able to use this feature on the operating tables?
The Operating Console only requires the Medical Scan to function like in the gift, and both Zeng-Hu Analyzer and Body Scanners will be able to print them for the console

>* On the operating table graphic, there is a little green animated pulse display. Could you make that change colour to red / whatever, when there is an audio alert condition? Currently it just stays the same even if the patient is dead.
Ooooh great idea, I'll try to add it in

>* The alarm text gets kinda spammy, and the chat spam prevention doesn't help because they are two different messages. You could change it to "Warning! x's Brain Activity and Blood Oxygenation are in critical condition." when both are true? Maybe increase the delay between messages (not necessarily the beeps aswell) too?
Mm! Absolutely adding the first one and tweaking the second accordingly. I was having trouble gauging how often it should sound off the alarm and went with more frequent than less.

**Comment by VisVirific** [comment]
Before I try to fix the icon conflict, here's the operating table icon

File: code/game/machinery/OpTable.dm [hunk 1]
```diff
@@ -82,11 +82,13 @@
 	if(!victim || !victim.lying || victim.loc != loc)
 		suppressing = FALSE
 		victim = null
-		var/mob/living/carbon/human/H = locate() in loc
-		if(istype(H))
-			if(H.lying)
-				icon_state = H.pulse() ? "[modify_state]-active" : "[modify_state]-idle"
-				victim = H
+	var/mob/living/carbon/human/H = locate() in loc
```
[the remark this answers is not in the document]
Comment (VisVirific) [reply code/game/machinery/OpTable.dm:85]: Got it. I'll see what I can do

[the remark this answers is not in the document]
Comment (VisVirific) [reply code/game/machinery/OpTable.dm]: Done! It should be working properly

File: code/game/machinery/OpTable.dm [hunk 2]
```diff
@@ -82,17 +82,19 @@
...
+		var/mob/living/carbon/human/H = locate() in loc
+		if(istype(H))
+			if(H.lying)
+				icon_state = H.pulse() ? "[modify_state]-active" : "[modify_state]-idle"
+				victim = H
 	if(victim && !victim.isSynthetic())
 		if(suppressing && victim.sleeping < 3)
 			victim.Sleeping(3 - victim.sleeping)
 			victim.willfully_sleeping = FALSE
+		if(ishuman(victim))
```
Comment (mikomyazaki) [inline code/game/machinery/OpTable.dm:94]: Victims cannot be non-human, so I don't think we need this line.

Rest is good though.

**Review by mikomyazaki — CHANGES_REQUESTED** [review]

File: code/game/machinery/OpTable.dm (hunk 2 above)
Comment (VisVirific) [reply code/game/machinery/OpTable.dm:94]: Unfortunately removing this line causes the operating table icon to not update. I do not know why

**Review by mikomyazaki — APPROVED** [review]

**Pull request merged**